### ISSUE TITLE
Alpha channel issues

### ISSUE THREAD

**Issue by chriseppstein** [wvanbergen/chunky_png#58]
Hi @wvanbergen,

We have this issue: https://groups.google.com/forum/#!topic/compass-users/U6xMb5INZhA

I suspect there's some way we can mitigate this in the compass spriting code. Can you please advise?

**Comment by jhamon** [comment]
I don't have an image that causes the problem described.  But after some investigation, I believe this issue resides in `chunky_png` and is at least partly the result of `transparent_color` being stored as a class variable on `ChunkyPNG::Image` before/during/after calls to `ChunkyPNG::Image.from_file`.  

`transparent_color` begins as `nil` on first load, but is not cleaned up after the loading of a file is completed.  If a file with a transparent_color is loaded, that setting will be applied to all subsequent file loads unless they specify a new transparent color.  Right now we can capture this in a test as:

```ruby
  context 'Decoding transparency' do
    after(:each) do
      # This fails right now
      expect(ChunkyPNG::Image.transparent_color).to be_nil
    end
    
    png_suite_files(:transparency, 'tp0*.png').each do |file|
      it "should not have transparency in #{File.basename(file)}" do
        expect(ChunkyPNG::Color.a(ChunkyPNG::Image.from_file(file)[0,0])).to eql 255
      end
    end

    # ... omitted tests ....
  end

```

This gives errors like 
```
1) PNG testuite Decoding transparency should have transparency in tbbn1g04.png
     Failure/Error: expect(ChunkyPNG::Image.transparent_color).to be_nil
       expected: nil
            got: 2004318207
     # ./spec/png_suite_spec.rb:86:in `block (3 levels) in <top (required)>'

  2) PNG testuite Decoding transparency should have transparency in tbbn2c16.png
     Failure/Error: expect(ChunkyPNG::Image.transparent_color).to be_nil
       expected: nil
            got: 2139062271
     # ./spec/png_suite_spec.rb:86:in `block (3 levels) in <top (required)>'
```

A quick-and-dirty fix might be to reset the value to nil after using it, but it seems to me like a property that should probably live on Canvas/Image instances since different images in memory may have different settings.  Same goes for `decoding_palette`.

What do you think about moving `transparent_color` and `decoding_palette` onto instances of `Image`? If you think it's a good idea, I can work on putting together a PR that does that.

**Comment by gmoigneu** [comment]
You'll find an image triggering this bug and the resulting sprite :
![p13](https://cloud.githubusercontent.com/assets/876016/7475540/7b2a4dd6-f348-11e4-9a4d-e60ebad87c02.png)
![gems-2x-s6a443179d4](https://cloud.githubusercontent.com/assets/876016/7475545/7fb0f4fe-f348-11e4-820f-78807c7f6b72.png)

```
chunky_png (1.3.4)
compass (1.0.3)
sass (3.4.13, 3.3.14)
```

**Comment by gmoigneu** [comment]
If that help some of you, the alpha bug happens to e only for png that have been optilized through ImageOptim before being sprited.

**Comment by mbrammer** [comment]
Same issue for me. I also user ImageOptim.

**Comment by wvanbergen** [comment]
Based on @jhamon's investigation, I opened a PR that may fix this issue: https://github.com/wvanbergen/chunky_png/pull/112.

@mbrammer @chriseppstein @jhamon @gmoigneu if you coudl test that branch, I would greatly appreciate it.

**Comment by wvanbergen** [comment]
Closing this, if it reoccurs, please re-open this issue.

**Comment by wvanbergen** [comment]
The fix should be part of ChunkyPNG 1.3.7
